saleor/plugins/webhook/tasks.py

```diff
@@ -82,17 +91,102 @@ def _get_webhooks_for_event(event_type, webhooks=None):
     return webhooks
 
 
-def trigger_webhooks_async(data, event_type, webhooks):
-    payload = EventPayload.objects.create(payload=data)
-    deliveries = create_event_delivery_list_for_webhooks(
-        webhooks=webhooks,
-        event_payload=payload,
-        event_type=event_type,
-    )
+def create_deliveries_for_subscriptions(
+    event_type, subscribable_object, webhooks, meta=None
+) -> List[EventDelivery]:
+    """Create webhook payload based on subscription query.
+
+    It uses a defined subscription query, defined for webhook to explicitly determine
+    what fields should be included in the payload.
+
+    :param event_type: event type which should be triggered.
+    :param subscribable_object: subscribable object to process via subscription query.
+    :return: List of event deliveries to send via webhook tasks.
+    """
+    if event_type not in SUBSCRIBABLE_EVENTS:
+        logger.info(
+            "Skipping subscription webhook. Event %s is not subscribable.", event_type
+        )
+        return []
+
+    context = initialize_context()
+    event_payloads = []
+    event_deliveries = []
+    for webhook in webhooks:
+        data = generate_payload_from_subscription(
+            event_type=event_type,
+            subscribable_object=subscribable_object,
+            subscription_query=webhook.subscription_query,
+            context=context,
+            app=webhook.app,
+        )
+        if not data:
+            logger.warning(
```

Comment:
Instead of single warning, don't you think, that we should build a EventDelivery object, that would allow user to see, that something was wrong with the query? Currently, they won't know that something was wrong before checking logs..

Reply:
You think about only single EventDelivery object that will have FAILED status - that will indicate that something went wrong but with no more info? Because creating related EventDeliveryAttempt that will have some more info about failure seems a bit off to me - cause it would not be a proper delivery attempt (no response and request headers etc.). And we need to keep in mind that creating EventDelivery would allow us to use `EventDeliveryRetry` mutation on that delivery.
I'll try to think of a better way to signal to user that payload generation went wrong.

Reply:
As we agreed - this is something to be further discussed and done in future PRs.